packages/app-builder-lib/src/asar/asarUtil.ts

```diff
@@ -45,237 +43,115 @@ export class AsarPackager {
-          return
-        }
+        matchUnpacker(file, dest, metadata)
+        taskManager.addTask(writeFileOrSymlink(transformedData, file, dest, metadata))
 
-        // https://github.com/yarnpkg/yarn/pull/3539
-        const stat = metadata.get(file)
-        if (stat != null && stat.size < 2 * 1024 * 1024) {
-          readFile(file)
-            .then(it => {
-              writeStream.write(it, () => w(index + 1))
-            })
-            .catch((e: any) => reject(`Cannot read file ${file}: ${e.stack || e}`))
-        } else {
-          const readStream = createReadStream(file)
-          readStream.on("error", reject)
-          readStream.once("end", () => w(index + 1))
-          readStream.on("open", () => {
-            readStream.pipe(writeStream, {
-              end: false,
-            })
-          })
+        if (taskManager.tasks.length > MAX_FILE_REQUESTS) {
+          await taskManager.awaitTasks()
         }
       }
-
-      writeStream.write(headerBuf, () => w(0))
-    })
-  }
-}
-
-async function order(filenames: Array<string>, orderingFile: string, src: string) {
-  const orderingFiles = (await readFile(orderingFile, "utf8")).split("\n").map(line => {
-    if (line.indexOf(":") !== -1) {
-      line = line.split(":").pop()!
     }
-    line = line.trim()
-    if (line[0] === "/") {
-      line = line.slice(1)
-    }
-    return line
-  })
-
-  const ordering: Array<string> = []
-  for (const file of orderingFiles) {
-    const pathComponents = file.split(path.sep)
-    for (const pathComponent of pathComponents) {
-      ordering.push(path.join(src, pathComponent))
+    await taskManager.awaitTasks()
+    return {
+      unpackedPaths: Array.from(unpackedPaths),
+      copiedFiles: Array.from(copiedFiles),
     }
   }
 
-  const sortedFiles: Array<string> = []
-  let missing = 0
-  const total = filenames.length
-  for (const file of ordering) {
-    if (!sortedFiles.includes(file) && filenames.includes(file)) {
-      sortedFiles.push(file)
+  private async copyFileOrData(data: string | Buffer | undefined, source: string, destination: string, stat: fs.Stats) {
+    await fs.mkdir(path.dirname(destination), { recursive: true })
+    if (data) {
+      await fs.writeFile(destination, data, { mode: stat.mode })
+    } else {
+      await fs.copyFile(source, destination)
     }
   }
-  for (const file of filenames) {
-    if (!sortedFiles.includes(file)) {
-      sortedFiles.push(file)
-      missing += 1
-    }
-  }
-  log.info({ coverage: ((total - missing) / total) * 100 }, "ordering files in ASAR archive")
-  return sortedFiles
-}
-
-function copyFileOrData(fileCopier: FileCopier, data: string | Buffer | undefined | null, source: string, destination: string, stats: Stats) {
-  if (data == null) {
-    return fileCopier.copy(source, destination, stats)
-  } else {
-    return writeFile(destination, data)
-  }
 }
 
 function orderFileSet(fileSet: ResolvedFileSet): ResolvedFileSet {
```

Comment:
Is this order function deletable? The electron/asar will reorder it, so this order seems a bit redundant.